### DISCUSSION TITLE
Ability to opt out of "No routes matched location" warnings

### DISCUSSION THREAD

**Discussion by AndrewLeedham** [remix-run/react-router#10851] [Proposals]
In Remix I have implemented a custom express integration where 404's fall-through to other express middleware. This works great, however we get warnings that react-router does not match any routes, which is expected and an un-needed warning in our case.

It would be great if we could opt out of this warning firing: https://github.com/remix-run/react-router/blob/4b494b935d62cd1244fe5c091db920d3f0315e9e/packages/react-router/lib/hooks.tsx#L430

Also, as a side not in regular development one log is easy enough to ignore. But in e2e testing it floods the output.

**Comment by cjones26** [comment]
I would like to add in order for us to upgrade a massive MFE app we had to use certain workarounds in order to upgrade to v6.  Until we can reorganize our routes, this warning is absolutely killing us, even though our routing is working properly.

**Reply by cjones26** [reply]
Note I was also able to at least band-aid this by adding the following at the bottom of each of my `<Routes>` components:

```
<Route path="*" element={null} />
```

Not a great solution, but it should work for our case right now until we can migrate to the data router.
